core/src/main/java/tc/oc/pgm/action/ActionParser.java

```diff
@@ -309,6 +313,18 @@ public ReplaceItemAction parseReplaceItem(Element el, Class<?> scope) throws Inv
     return new ReplaceItemAction(matcher, item, keepAmount, keepEnchants);
   }
 
+  @MethodParser("timelimit")
+  public TimeLimitAction parseTimeLimit(Element el, Class<?> scope) throws InvalidXMLException {
+    Duration duration = XMLUtils.parseDuration(el.getAttribute("duration"), Duration.ZERO);
```

Comment:
Reuse the parsing code in time limits, and just pass in a TimeLimit to the action